test/functional/sqljs/auto-save.ts

```diff
@@ -0,0 +1,85 @@
+import "reflect-metadata";
+import {expect} from "chai";
+import {createConnection} from "../../../src/index";
+import {Post} from "./entity/Post";
+
+describe("sqljs driver > autosave", () => {
+    it("should call autoSaveCallback on insert, update and delete", async () => {
+        let saves = 0;
+        const callback = (database: Uint8Array) => {
+            saves++;
+        };
+
+        let connection = await createConnection({
+            type: "sqljs",
+            entities: [Post],
+            synchronize: true,
+            autoSaveCallback: callback,
+            autoSave: true
+        });
+        
+        const repository = connection.getRepository(Post);
+        let post = new Post();
+        post.title = "A post";
+        await repository.save(post);
+
+        let posts = [
+            {
+                title: "second post"
+            },
+            {
+                title: "third post"
+            }
+        ];
+
+        repository.createQueryBuilder("api").insert().into(Post).values(posts).execute();
```

Comment:
The only problem left is the autoSave detection. Right now autoSave is called when a transaction is committed or `insert()`, `update()` or `delete()` in the query runner is called and no transaction is active.
But this will not detect bulk inserts like this one as they are executed using the `query()` method from query runner.
Any ideas?

Reply:
maybe for now you simply call autoSave method of the driver in the case if dirver is sqlitejs right in the insert/update/delete query builders? Now Im not sure about better solution

Reply:
I don't know if this is that much better than my original "monkey patch" where I checked if the query contains a `INSERT`, `UPDATE` or `DELETE`